core/src/main/java/org/mule/runtime/core/internal/construct/AbstractPipeline.java

```diff
@@ -222,26 +222,40 @@ public InternalEvent process(InternalEvent event) throws MuleException {
     initialiseIfInitialisable(pipeline);
   }
 
+  /*
+   * Processor that dispatches incoming source Events to the internal pipeline the Sink. The way in which the Event is dispatched
+   * and how overload is handled depends on the Source back-pressure strategy.
+   */
   private ReactiveProcessor dispatchToFlow(Sink sink) {
     if (source.getBackPressureStrategy() == WAIT) {
-      return publisher -> from(publisher).doOnNext(event -> sink.accept(event))
+      // If back-pressure strategy is WAIT then using blocking `accept(Event event)` to dispatch Event
+      return publisher -> from(publisher)
+          .doOnNext(event -> {
+            try {
+              sink.accept(event);
+            } catch (RejectedExecutionException ree) {
+              MessagingException me = new MessagingException(event, ree, this);
+              event.getContext().error(exceptionResolver.resolve(me, getMuleContext()));
+            }
+          })
           .flatMap(event -> Mono.from(event.getContext().getResponsePublisher()));
     } else {
+      // If back-pressure strategy is FAIL/DROP then using back-pressure aware `accept(Event event)` to dispatch Event
       return publisher -> from(publisher).flatMap(event -> {
         if (sink.emit(event)) {
           return event.getContext().getResponsePublisher();
         } else {
           if (source.getBackPressureStrategy() == DROP) {
+            // If Event is not accepted and the back-pressure strategy is DROP then drop Event.
             return empty();
           } else {
-            RejectedExecutionException rejectedExecutionException = new RejectedExecutionException(format(OVERLOAD_ERROR_MESSAGE,
-                                                                                                          AbstractPipeline.this
-                                                                                                              .getName()));
+            // If Event is not accepted and the back-pressure strategy is FAIL then respond to Source with an OVERLOAD error.
+            RejectedExecutionException rejectedExecutionException =
```

Comment:
no subclass then?

Reply:
I asked, in a comment, why you thought this was valuable, maybe comment got lost. The reason i don't see much value is because Error is the primary way of declaring and propagating errors not, not the exception so I don't see value in sub-class personally.